Lib/inspect.py

```diff
@@ -396,7 +396,9 @@ def iscoroutinefunction(obj):
 
     Coroutine functions are defined with "async def" syntax.
     """
-    return _has_code_flag(obj, CO_COROUTINE)
+    return _has_code_flag(obj, CO_COROUTINE) or (
```

Comment:
Right now the function is called `iscoroutinefunction`. But, if applied to a callable object with `__call__` marked as `@markcoroutine` it will return `True`.

I might be wrong, but this does not seem like a matching semantics.

Other similar functions do not do that.

Reply:
@sobolevn — OK, I might need some direction then on the way forward. 🤔 

Despite the name — `iscoroutinefunction` — class instances with either an `async def __call__` or a `def __call__` where that has been marked with the new decorator — where we assume the writer knew what they were doing —  **should** return `True` here.